src/reporter/JestReporter.ts

```diff
@@ -0,0 +1,248 @@
+                origin: "", // TODO: figure out how to get the stack
+                type: "info",
+            });
+        });
+    }
+
+    /**
+     * Callback for when a test run is starting.
+     * @param numSuites The number of suites in the run
+     */
+    public onRunStart(numSuites: number) {
+        this.aggregatedResults.startTime = Date.now();
+        this.aggregatedResults.numTotalTestSuites = numSuites;
+
+        this.reporters.forEach((reporter) => {
+            reporter.onRunStart(this.aggregatedResults, {
+                estimatedTime: 0, // TODO: estimate this maybe?
+                showStatus: false,
+            });
+        });
+    }
+
+    /**
+     * Callback for when all test files have been executed.
+     */
+    public onRunComplete() {
+        let contextSet = new Set([createContext(this.projectConfig)]);
+        this.reporters.forEach((reporter) => {
+            reporter.onRunComplete(contextSet, this.aggregatedResults);
+        });
+    }
+
+    /**
+     * Callback for when brs encounters an execution error.
+     * @param filename The name of the file that failed
+     * @param index The TAP index of the file
+     * @param reason The error that was thrown to cause the execution error
+     */
+    public onFileExecError(
+        filename: string,
+        index: number,
+        reason: Error | BrsError | BrsError[]
+    ) {
+        // If we get an array of errors, report the first one.
+        reason = Array.isArray(reason) ? reason[0] : reason;
+
+        if (isBrsError(reason)) {
+            this.currentResults.testExecError = {
+                stack: `\nat ${reason.location.file}:${reason.location.start.line}:${reason.location.start.column}`,
+                message: reason.message,
+            };
+        } else {
+            this.currentResults.testExecError = {
+                stack: reason.stack,
+                message: reason.message,
+            };
+        }
+    }
+
+    /**
+     * Callback for when a file starts test execution. Creates empty results
+     * for the new file and sets it as the current results for this parser.
+     * @param filePath The path to the file that is starting its run.
+     */
+    public onFileStart(filePath: string) {
+        this.currentResults = createEmptyTestResult();
+        this.currentResults.testFilePath = path.join(process.cwd(), filePath);
+        this.currentResults.perfStats.start = Date.now();
+    }
+
+    /**
+     * Callback for when a file ends execution. Adds the results from the current
+     * file to the aggregated results, and tells each Jest reporter that this file has completed.
+     */
+    public onFileComplete() {
+        // Flatten console output because tap-parser splits output by newline in the "extra" event.
+        // We don't know which lines are supposed to go together, so just print them all together.
+        if (this.currentResults.console) {
+            this.currentResults.console = [
```

Comment:
This was the best solution I could think of. It's a little wonky, but it's not too far off from how Jest handles `console.log` statements. The main problem is we don't have any stack trace for the logs.

Jest reporter for `roca` (what this PR implements):
![Screen Shot 2021-03-17 at 1 09 16 PM](https://user-images.githubusercontent.com/11684087/111531937-62bb1e00-8722-11eb-9c1d-259e914c7d9e.png)

Actual jest reporter for JS (I put some `console.log`s in the `roca` project's JS test suite):
![Screen Shot 2021-03-17 at 1 11 47 PM](https://user-images.githubusercontent.com/11684087/111532053-841c0a00-8722-11eb-90ac-1567deb04286.png)

Reply:
I think this is fine as long as we get stack traces for errors which seems like we do already 😄

Reply:
Agreed!  Since we treat `brs` (specifically its `print` statements) as an opaque box from the `roca` perspective, I'm fine with losing that for now.  It might be possible to add an `eventEmitter` that handles writes to stdout instead of calling `interpreter.stdout.write()` directly from within `brs`'s implementation of `print`, but that's not necessary here at all.  It'd be neat to consider for the future though — especially in combination with a `--quiet` in `brs` to suppress writes to `stdout`?

Reply:
>  It'd be neat to consider for the future though — especially in combination with a --quiet in brs to suppress writes to stdout?

I agree, that'd be great!